### DISCUSSION TITLE
Make caching location configurable

### DISCUSSION THREAD

**Discussion by hveiga** [nginxinc/nginx-s3-gateway#258] [Ideas]
Hi,

I was wondering if it would make sense to make the paths where nginx is caching requests configurable. I think this could be beneficial in a container setting where you want to store the cached requests in a different volume.

Right now, the path is hardcoded:

https://github.com/nginxinc/nginx-s3-gateway/blob/a2f1540e76595fe5669fb5d373c4a03b35a79884/common/etc/nginx/templates/cache.conf.template#L3

https://github.com/nginxinc/nginx-s3-gateway/blob/a2f1540e76595fe5669fb5d373c4a03b35a79884/common/etc/nginx/templates/cache.conf.template#L11

One caveat I could see about storing the cached requests somewhere else is if the container gets restarted, those files might be left behind and never cleaned up by nginx. Is there any other scenario where this could be a problem?

After the discussion, and if there is an agreement, I am happy to open a PR to introduce new environment variable to configure these paths.

Thank you!

**Comment by 4141done** [comment]
Thanks for bringing this up @hveiga - I think this sounds like a good idea generally. Most of the potential issues sound like ones that you should consider yourself if setting up a custom caching location:
* Orphaned cache files
* Available space
* Slowness or strange behavior of network stores

What do you think should be the behavior if the configuring location doesn't exist or is not writable?  NGINX will fail to start by default in that situation:
```
nginx: [emerg] mkdir() "/bad_location" failed (30: Read-only file system)
```

Hope other users can comment on this as well 😅

**Comment by dekobon** [comment]
I'm a little confused about this request. Although you can't change the path of the cache directory `/var/cache/nginx/s3_proxy`, you can still use docker to mount a volume into its path by doing `-v mycachepath:/var/cache/nginx/s3_proxy`. Shouldn't that accomplish what you need to do?

**Reply by hveiga** [reply]
Honestly, that option did not occurred to me and I think it would be sufficient to accomplish what I am looking for. I'll give it a try and report back. Thank you for the suggestion.
